server/service/appconfig.go

```diff
@@ -728,6 +728,12 @@ func (svc *Service) validateMDM(
 			return
 		}
 	}
+	if !svc.config.MDM.IsMicrosoftWSTEPSet() {
+		if mdm.WindowsEnabledAndConfigured {
+			invalid.Append("mdm.windows_enabled_and_configured", "Windows WSTEP configuration must be provided to enable Windows MDM")
```

Comment:
This is pending confirmation of the copy: https://github.com/fleetdm/fleet/issues/14446#issuecomment-1789502397

So let's not merge yet even if approved, and also the frontend needs to be modified to display the server's error message when the status is 422 (currently it displays a generic "Unable to turn on Windows MDM, please try again" message).